### PR TITLE
Better gRPC error on context error from CommitStatus service

### PR DESCRIPTION
**Pull request by bestbeforetoday**
Rather than always return a FailedPrecondition error on failure obtaining commit status, return either a DeadlineExceeded or Canceled error on a context error. This may happen if the call is cancelled from the client end, either by an explicit context cancel or timeout.

### CODE REVIEW

File: internal/pkg/gateway/apiutils.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func getChannelAndChaincodeFromSignedProposal(signedProposal *peer.SignedProposa
 	return channelHeader.ChannelId, spec.ChaincodeSpec.ChaincodeId.Name, len(payload.TransientMap) > 0, nil
 }
 
-func rpcError(code codes.Code, message string, details ...proto.Message) error {
+func newRpcError(code codes.Code, message string, details ...proto.Message) error {
```
Comment (ale-linux) [inline internal/pkg/gateway/apiutils.go:48]: Out of curiosity, why the change of function name?

**Review by ale-linux — APPROVED** [review]

File: internal/pkg/gateway/apiutils.go (hunk 1 above)
Comment (bestbeforetoday) [reply internal/pkg/gateway/apiutils.go:48]: I introduced a `toRpcError` function and just wanted the names to be a little more distinct to avoid confusion
